### PR TITLE
Div 1956 invalid case ref hitting gov pay

### PR DESCRIPTION
**Pull request by niyioyelade**
Throwing an error when Case ID is not provided.

https://tools.hmcts.net/jira/browse/DIV-1956

Promise rejects when case ID/reference is not provided
Error is logged in this scenario as well
Unit tests updated to cover this scenario
This will prevent payments with invalid case ID being forwarded to prod Gov Pay subsequently

### CODE REVIEW

File: app/steps/pay/pay-online-only/index.test.js [hunk 1]
```diff
@@ -188,7 +204,7 @@ describe(modulePath, () => {
         it('redirects to the gov.uk payment page', done => {
           // Act.
           const featureMock = featureTogglesMock
-            .when('onlineSubmission', true, testCustom, agent, underTest, cookies, response => {
+            .when('onlineSubmission', true, testCustom, agent, underTest, [], response => {
```
Comment (Rob-Doherty) [inline app/steps/pay/pay-online-only/index.test.js]: Is this a test that not including cookies should still redirect to payments page? Is this definitely correct?

File: app/steps/pay/pay-online-only/index.js [hunk 2]
```diff
@@ -84,6 +84,11 @@ module.exports = class PayOnline extends Step {
     const caseId = req.session.caseId;
     const siteId = get(req.session, `court.${req.session.courts}.siteId`);
 
+    if (!caseId) {
+      logger.error('Case ID is missing');
+      res.redirect('/generic-error');
```
Comment (dtaylor7) [inline app/steps/pay/pay-online-only/index.js]: needs to be `return res.redirect('/generic-error');`

so as to not execute the rest of the code in this function

Comment (niyioyelade) [reply app/steps/pay/pay-online-only/index.js]: return statement added. Thanks

File: app/steps/pay/pay-online-only/index.test.js (hunk 1 above)
Comment (niyioyelade) [reply app/steps/pay/pay-online-only/index.test.js]: This is now sorted. Cookies has been passed in. Thanks.

File: app/steps/pay/pay-online-only/index.test.js [hunk 3]
```diff
@@ -107,34 +107,61 @@ describe(modulePath, () => {
...
+
+        withSession(done, agent, session);
+      });
+      it('redirects to the generic error page', done => {
+        // Act.
+        const featureMock = featureTogglesMock
+          .when('onlineSubmission', true, testCustom, agent, underTest, cookies, response => {
+            // Assert.
+            expect(response.status).to.equal(statusCodes.MOVED_TEMPORARILY);
+            expect(response.header.location).to.equal('/generic-error');
```
Comment (dtaylor7) [inline app/steps/pay/pay-online-only/index.test.js]: would add test here to ensure that rest of code isn't ran... i.e. `expect(serviceTokenService.setup.called).to.eql(false);`

Comment (niyioyelade) [reply app/steps/pay/pay-online-only/index.test.js]: Good spot. Done.

**Pull request merged**